File: teoricos/arvores_binarias/ArvoreC/ArvoreBinariaDeLista/no.c
#include <stdlib.h>
#include "no.h"
#include "lista.h"

No* novoNo(char elemento) {
    No* novo = (No*) malloc(sizeof(No));
    novo->elemento = elemento;
    novo->esq = NULL;
    novo->dir = NULL;
    return novo;
}

File: teoricos/balanceamento_arvores/alvinegra/No.java
class No {
    // Cor do Nó
    public boolean cor;
    // Elemento do nó
    public int elemento;
    public No esq, dir;
    
    public No() {
        this(-1);
    }

    public No(int elemento) {
        this(elemento, false, null, null);
    }

    public No(int elemento, boolean cor) {
        this(elemento, cor, null, null);
    }

    public No(int elemento, boolean cor, No esq, No dir) {
        this.cor = cor;
        this.elemento = elemento;
        this.esq = esq;
        this.dir = dir;
    }
}